app/services/mixtape_maker.rb

```diff
@@ -0,0 +1,15 @@
+class MixtapeMaker
+
+  def self.generate(no_song = 5, artists:)
+    new.generate(no_song, artists: artists)
+  end
+
+  def generate(no_song = 5, artists:)
+    if artists.blank?
```

Comment:
So, there is no way that `artists` here is going to be nil (due to the named argument), but right, it could be empty.